### PR TITLE
chore: Bump Rust, Go and Node dependencies

### PR DESCRIPTION
**Pull request by Techassi**
# Description

This PR bumps various Rust and Node dependencies. Should resolve #185, #211, #221, #225, #229 and #230.

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

### CODE REVIEW

**Comment by Techassi** [comment]
I will include the `setup-go` action in other workflows as well, so that we can more easily control which Go version is used (instead of using the one provided by the image).

**Comment by Techassi** [comment]
The "Publish Docker Image" step fails, because some Nix stuff fails. I'm out of my depth to determine what the exact issue is. Maybe @nightkr or @NickLarsenNZ can take a look.

**Comment by nightkr** [comment]
I can take a look.

**Comment by nightkr** [comment]
Tilt is happy as of ba1e2a78f517d602ca7e714b09c76cd535636849, hopefully GitHub should be too.

File: Cargo.toml [hunk 1]
```diff
@@ -53,9 +53,9 @@ serde_json = "1.0"
 serde_yaml = "0.9"
 sha2 = "0.10"
 snafu = { version = "0.8", features = ["futures"] }
-stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "stackable-operator-0.67.0" }
+stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "stackable-operator-0.67.1" }
 tera = "1.18"
-termion = "3.0.0"
+termion = "4.0.0"
```
Comment (NickLarsenNZ) [inline Cargo.toml]: Apparently we should be using major.minor...

```suggestion
termion = "4.0"
```

File: default.nix [hunk 2]
```diff
@@ -42,6 +42,23 @@ rec {
...
+        # so we download it instead, and put it where it expects to be cached
+        preConfigure =
+          let
+            swaggerUi = pkgs.fetchurl {
+              url = "https://github.com/swagger-api/swagger-ui/archive/refs/tags/v5.17.3.zip";
+              hash = "sha256-zrb8feuuDzt/g6y7Tucfh+Y2BWZov0soyNPR5LBqKx4=";
+            };
+          in ''
+            mkdir -p target/build/utoipa-swagger-ui.out
+            ln -s ${swaggerUi} target/build/utoipa-swagger-ui.out/swagger-ui.zip
```
Comment (NickLarsenNZ) [inline default.nix]: I assume the path is guaranteed to not contain spaces, but...

```suggestion
            ln -s "${swaggerUi}" target/build/utoipa-swagger-ui.out/swagger-ui.zip
```

File: yarn.lock
Comment (NickLarsenNZ) [inline yarn.lock:1]: Maybe we can add yarn.lock to .gitattributes...


```gitattributes
yarn.lock linguist-generated
```

File: go.sum
Comment (NickLarsenNZ) [inline go.sum:1]: Maybe we can add go.sum to .gitattributes...


```gitattributes
go.sum linguist-generated
```

**Review by NickLarsenNZ — DISMISSED** [review]
LGTM, left comments which are optional to action now.

File: yarn.lock
Comment (Techassi) [reply yarn.lock:1]: Oh yeah, we can do that. Same applies for the `go.sum` file.

Comment (Techassi) [reply yarn.lock:1]: Added in 9cfb1c5

File: go.sum
Comment (Techassi) [reply go.sum:1]: Added in 9cfb1c5

File: default.nix (hunk 2 above)
Comment (Techassi) [reply default.nix]: I assume this should be fine. Any objections @nightkr?

File: Cargo.toml (hunk 1 above)
Comment (NickLarsenNZ) [reply Cargo.toml]: I assume it was done. Mobile view requires more effort to see that.

Comment (NickLarsenNZ) [reply Cargo.toml]: 485611ae77c22e9c6e3ddcdc18891c067c3a5a58

**Review by NickLarsenNZ — DISMISSED** [review]

File: default.nix (hunk 2 above)
Comment (nightkr) [reply default.nix]: Sure, lgtm.

Comment (Techassi) [reply default.nix]: Alright.

**Review by NickLarsenNZ — APPROVED** [review]
Perfection 👌

**Comment by sbernauer** [comment]
Recalling my last big Go bump you lucky ones :) Nice!

**Pull request merged**